tools/automated-dns-for-instances/README.md

```diff
@@ -0,0 +1,126 @@
+# Automating Cloud DNS for VM Instances Tool
+
+## 1. Overview
+Using this document you will learn how to automate the creation and deletion of DNS ‘A’ records for compute instances on Google Cloud Platform.  We will use several GCP services to monitor virtual machine changes and act on specific events during their lifecycle. 
+
+The procedure involves the following steps:
+
+1. Creation of service account(s)
+2. Enable & Configure Cloud Asset Inventory (CAI) and Pub/Sub
+3. Deploy Cloud Function and Pub/Sub Trigger
+
+We will use the command line gcloud utility to configure and deploy all of the services. Installation for various environments can be found at https://cloud.google.com/sdk/install. Once installed you will need to authenticate & configure it:
+
+```
+gcloud auth login
+```
+
+This will open a browser window and you will log in using your existing GCP user account. Once complete you can list the current gcloud configuration:
+
+```
+gcloud config list
+```
+
+### Example output:
+
+```shell
+[compute]
+region = us-central1
+zone = us-central1-a
+[core]
+account = user@domain
+disable_usage_reporting = True
+project = a_project_id
+
+Your active configuration is: [default]
+```	
+
+Update the project, region and zone to reflect where you want to deploy the resources you will provision. We will use variables set in prior code blocks so stay in the same terminal for doing all of the work.
+
+## 2. Create Service Accounts
+You will need one or more service accounts for integrating the various services. You can create a single account and use it across Cloud Asset Inventory & Cloud Functions or create two separate ones. In this example we will create two separate accounts.
+
+### 2.1 SA Creation
```

Comment:
Would prefer to see Terraform for all these bits.
Less error prone / better user experience than gcloud commands.

Reply:
See @kardiff18's [similar asset](https://github.com/GoogleCloudPlatform/professional-services/tree/master/examples/ml-audio-content-profiling) for example